floodviz/static/js/hydrograph.js

```diff
@@ -31,91 +31,114 @@ document.addEventListener("DOMContentLoaded", function (event) {
 			return y(d.value);
 		});
 
-	// Adds the svg canvas
-	var svg = d3.select("#hydrograph")
-		.append("svg")
-		.attr("width", width + margin.left + margin.right)
-		.attr("height", height + margin.top + margin.bottom)
-		.append("g")
-		.attr("transform",
-			"translate(" + margin.left + "," + margin.top + ")");
-
 	// Get the data
 	d3.json(data_path, function (error, data) {
-		data.forEach(function (d) {
-			d.value = Number(d.value);
-		});
 
-		// Scale the range of the data
-		x.domain(d3.extent(data, function (d) {
-			return d.time_mili;
-		}));
-		y.domain([20, d3.max(data, function (d) {
-			return d.value;
-		})]);
+		//save the original data
+		var original_data = data;
+
+		update(data);
+
+		function update(data) {
+
+			// Adds the svg canvas
+			var svg = d3.select("#hydrograph")
+				.append("svg")
+				.attr("width", width + margin.left + margin.right)
+				.attr("height", height + margin.top + margin.bottom)
+				.append("g")
+				.attr("transform",
+					"translate(" + margin.left + "," + margin.top + ")");
+
+			data.forEach(function (d) {
```

Comment:
Consider using map rather than forEach. It would be clearer that you are actually creating a new array by transforming the original-data. For example  use data = original_data.map(function)...